src/analysis/souffle/authorization_logic.dl

```diff
@@ -52,7 +54,29 @@ isPrincipal(speaker) :- saysDowngrades(speaker, _, _).
 //-----------------------------------------------------------------------------
 downgrades(path, tag) :- ownsTag(owner, tag), saysDowngrades(owner, path, tag).
 
+claimNotHasTag(path, tag) :- downgrades(path, tag).
+
+// This rule conservatively says that if _any_ principal claims that a path has
+// a tag, then it has the tag according to the taint analysis. This rule is
+// likely to be adjusted later. For confidentiality tags this should be OK to
+// start with.
+claimHasTag(path, tag) :- saysHasTag(_, path, tag).
```

Comment:
Don't we need check the `ownsTag`? Otherwise, anyone can add a claim?

Reply:
Checking ownership here does sound reasonable at a glance. Although, since in this MVP we just have confidentiality tags, allowing anyone to add a tag is harmless, so we don't strictly need that yet. I can add the ownsTag check anyway since it does sound reasonable.